src/client/features/view/components/menu/search.js

```diff
@@ -0,0 +1,96 @@
+const _ = require('lodash');
+
+//Apply hover styling to a collection of nodes
+function updateStyling(style, matchedNodes) {
+  matchedNodes.forEach(node => {
+    let isCompartment = node.data('class') === 'compartment';
+
+    //Collapse Expanded Targets
+    if (node.isExpanded() && !isCompartment) { node.collapse(); }
+
+    //Apply style to the parent
+    if (node.isChild() && node.parent().data('class') !== 'compartment') { node = node.parent(); }
+
+    applySearchStyle(node, style);
+  });
+}
+
+//Determine if a regex pattern is valid
+function validateRegex(pattern) {
+  var parts = pattern.split('/'),
+    regex = pattern,
+    options = "";
+  if (parts.length > 1) {
+    regex = parts[1];
+    options = parts[2];
+  }
+  try {
+    let regexObj = new RegExp(regex, options);
+    return regexObj;
+  }
+  catch (e) {
+    return null;
+  }
+}
+
+function applySearchStyle(eles, style) {
+  eles.style(style);
+}
+
+//Search for nodes that match an entered query
+function searchNodes(query, cy) {
+  const searchValue = query.target.value;
+  const isBlank = _.isString(searchValue) ? !!_.trim(searchValue) : false;
+  const isRegularExp = _.startsWith(searchValue, 'regex:') && validateRegex(searchValue.substring(6));
+  const isExact = _.startsWith(searchValue, 'exact:');
+
+  let visibleNodes = Array.prototype.slice.call(cy.nodes(), 0);
```

Comment:
What does this line do?

Reply:
I have simplified line 47-50 into just 2 lines to make the code a lot easier to read and understand. This line is no longer needed.